Description: Upstream changes introduced in version 1.5.3-2.11
 This patch has been created by dpkg-source during the package build.
 Here's the last changelog entry, hopefully it gives details on why
 those changes were made:
 .
 cups (1.5.3-2.11) unstable; urgency=low
 .
   [ Julien Cristau ]
   * In the dbus UTF8 checker:
     - Use C.UTF-8 instead of not-certainly-available en_US.UTF-8; make
       cups depend on libc-bin (>= 2.13).
     - Make sure the buffer is really big enough.
 .
 The person named in the Author field signed this changelog entry.
Author: Didier Raboud <odyx@debian.org>

---
The information above should follow the Patch Tagging Guidelines, please
checkout http://dep.debian.net/deps/dep3/ to learn about the format. Here
are templates for supplementary fields that you might want to add:

Origin: <vendor|upstream|other>, <url of original patch>
Bug: <url in upstream bugtracker>
Bug-Debian: http://bugs.debian.org/<bugnumber>
Bug-Ubuntu: https://launchpad.net/bugs/<bugnumber>
Forwarded: <no|not-needed|url proving that it has been forwarded>
Reviewed-By: <name and email of someone who approved the patch>
Last-Update: <YYYY-MM-DD>

--- cups-1.5.3.orig/notifier/dbus.c
+++ cups-1.5.3/notifier/dbus.c
@@ -189,11 +189,11 @@ validate_utf8 (const char *str)
 
   /* Make sure our buffer is at least as large as the input string */
   str_len = strlen (str);
-  if (str_len >= buflen)
+  if (str_len > buflen)
   {
     if (buffer == NULL)
       /* Set encoding type to UTF-8 the first time we need to */
-      setlocale (LC_CTYPE, "C.UTF-8");
+      setlocale (LC_CTYPE, "en_US.UTF-8");
 
     buflen = str_len + 1;
     buffer = realloc (buffer, buflen);
